A27-xds-global-load-balancing.md

```diff
@@ -0,0 +1,425 @@
+A27: xDS-Based Global Load Balancing
+----
+* Author(s): Mark D. Roth
+* Approver: a11r
+* Status: Draft
+* Implemented in: (in progress in C-core, Java, and Go)
+* Last updated: 2020-01-07
+* Discussion at: https://groups.google.com/d/topic/grpc-io/GWIH4XhIqHA/discussion
+
+## Abstract
+
+gRPC currently supports its own ["grpclb"
+protocol](https://github.com/grpc/grpc-proto/blob/master/grpc/lb/v1/load_balancer.proto)
+for [look-aside
+load-balancing](https://github.com/grpc/grpc/blob/master/doc/load-balancing.md).
+However, the popular [Envoy](http://envoyproxy.io) proxy uses the [xDS
+API](https://github.com/envoyproxy/data-plane-api/tree/master/envoy/api/v2)
+for many types of configuration, including load balancing, and that API
+is evolving into a standard that will be used to configure a variety of
+data plane software.  In order to converge with this industry trend, gRPC
+will be moving from its original grpclb protocol to the new xDS protocol.
+
+The xDS protocol allows configuring a variety of features, including
+many that gRPC does not currently support.  Over time, we will be slowly
+adding support for more of these features in gRPC.  We will publish a
+gRFC for each set of xDS features as we add support for them.
+
+This document describes the changes we are making to support global load
+balancing features, which is the set of xDS functionality that we are
+initially targetting.  This affects the gRPC client only; use of xDS in
+the gRPC server will come later as part of a separate set of xDS features.
+
+## Background
+
+The xDS API is actually a suite of APIs with names of the form "x
+Discovery Service", where there are many values for "x" (hence the name
+"xDS" for the overall protocol suite).  There is not currently a formal
+spec for the protocol, but the best reference is at
+https://www.envoyproxy.io/docs/envoy/latest/api-docs/xds_protocol.
+Readers of this document should familiarize themselves with the xDS API
+before proceeding.
+
+The xDS APIs are essentially pub/sub mechanisms, where each API allows
+subscribing to a different type of configuration resource.  In the xDS
+API flow, the client uses the following main APIs, in this order:
+
+- __Listener Discovery Service (LDS)__: Returns `Listener` resources.
+  Used basically as a convenient root for the gRPC client's configuration.
```

Comment:
"gRPC client" sometimes can be ambiguous (at least to me). It can mean a client of the xDS management server or a data plane client of its data plane server (which is probably the meaning here). Wonder if there's a way to clarify this (but may not be a problem to others).

Reply:
The client of the xDS management server is referred to here as the XdsClient.  The term "gRPC client" means the data plane client.

I think this is pretty clear as-is, but feel free to suggest alternative wording to improve it.